homeassistant/components/nasweb/config_flow.py

```diff
@@ -0,0 +1,132 @@
+"""Config flow for NASweb integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+from webio_api import WebioAPI
+from webio_api.api_client import AuthError
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN, NASWEB_SCHEMA_IMG_URL
+from .coordinator import NASwebCoordinator
+from .nasweb_data import NASwebData
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate user-provided data."""
+    webio_api = WebioAPI(data[CONF_HOST], data[CONF_USERNAME], data[CONF_PASSWORD])
+    if not await webio_api.check_connection():
+        raise CannotConnect
+    try:
+        await webio_api.refresh_device_info()
+    except AuthError as e:
+        raise InvalidAuth from e
+
+    nasweb_data = NASwebData()
+    nasweb_data.initialize(hass)
```

Comment:
How is the case where `get_url` raises `NoURLAvailableError` handled?

Reply:
At the time I missed the fact that it could raise error. Now it's catched in config flow and in `async_setup_entry`.